src/esi/Parser.cc

```diff
@@ -25,19 +27,29 @@ ESIParser::GetRegistry()
 ESIParser::Pointer
 ESIParser::NewParser(ESIParserClient *aClient)
 {
-    if (Parser == NULL) {
-        Parser = GetRegistry().front();
-
-        // if type name matters, use it
+    if (!Parser) {
+        const char *use;
         if (strcasecmp(Type, "auto") != 0) {
-            for (auto *p : GetRegistry()) {
-                if (p && strcasecmp(p->name, Type) != 0)
-                    Parser = p;
-            }
+            // if esi_parser is configured, use that
+            use = Type;
+        } else {
+#if HAVE_LIBXML2
+            // libxml2 is the more secure. prefer when possible
+            use = "libxml2";
+#else
+            // expat is more widely available
+            use = "expat";
+#endif
+        }
+
+        for (auto *p : GetRegistry()) {
+            if (p && strcasecmp(p->name, use) == 0)
+                Parser = p;
         }
 
-        if (Parser == NULL)
-            fatal ("Unknown ESI Parser type");
+        if (!Parser)
+            fatalf("Unknown ESI Parser type '%s'", use);
+        debugs(86, 2, "Starting " << Parser->name << " ESI parser.");
```

Comment:
This is here with the "Started" prefix to fit in with the other startup cache.log lines.

Reply:
I do not think we should standardize a "Started" prefix for _everything_ we initialize on startup. Nothing is really "started" here, and other phrases can describe what is going on much better (one specific replacement was [suggested](https://github.com/squid-cache/squid/pull/968/files#r791354514), but I do not insist on it).

I do not think we should use periods, especially in level-2+ messages, because appending them costs a lot in terms of code noise and debugging performance in many use cases.

I do not think we should use capitalization for level-2+ messages because it often makes the combination of the function name and the following message text a bit less readable while many messages (correctly) use/rely on that function name being essentially a part of the message phrase (for code noise reduction and debugging performance reasons).

Reply:
"we should" which apply to a wide range of Squid features and components are out of scope here. There are 1 debugs which matches your requested "should" and 12 which use this implemented sentence style.

They are not specifically limited to the mainStartup() sequence. The consistent thing with timing is that they occur on feature initializing its global state. Here it is initializing ESI the parser singleton to use for the remainder of Squid lifetime.

Reply:
> "we should" which apply to a wide range of Squid features and components are out of scope here. There are 1 debugs which matches your requested "should" and 12 which use this implemented sentence style.

The correct scope for my statements is the whole Squid -- the statements describe the goal Squid code should be moving towards when it comes to level-2+ debugs() messages about module/feature initialization. Within that scope, there are probably about 20-60 applicable debugs(), depending on whether we focus on module initialization (~20) or major feature configuration (~60). One cannot find applicable debugs() quickly because many of them are not using the flawed "Starting..." pattern[^1] -- the "wrongs" differ a lot!

You are right that this PR only has one in-scope statement -- the debugs() this PR is _adding_. I see no good reason to follow a flawed pattern when adding this new statement because there is no consistency here to _preserve_ -- these module/feature initialization statements are all over the place already. It would be better to do this debugs() correctly, then to do it wrong and then fix it later.

Again, I do not insist on any changes here. Just outlining where I think we should be moving towards, eventually. I realize that there may even be no consensus regarding those goals.

[^1]: I counted 8 of `Starting...` debugs(), and only **one** of them is a level-2 debugs() that we are discussing here:
    $ git grep 'debugs.*"Starting' upstream/master | wc -l
    8
    $ git grep 'debugs.*"Starting' upstream/master | fgrep -v DBG_ | wc -l
    1
    $